hack/ginkgo-e2e.sh

```diff
@@ -79,11 +79,28 @@ else
     )
 fi
 
+# Set the NODE_INSTANCE_GROUP based on NODE_INSTANCE_PREFIX. 
+# The NODE_INSTANCE_PREFIX and NODE_INSTANCE_GROUP vars are used the 'if' block below the
+# when the provider is gce.
 if [[ -n "${NODE_INSTANCE_PREFIX:-}" ]]; then
   NODE_INSTANCE_GROUP="${NODE_INSTANCE_PREFIX}-group"
 fi
 
-if [[ "${KUBERNETES_PROVIDER}" == "gce" ]]; then
+# If the NODE_INSTANCE_GROUP is not set, and the provider is gce run this code block.
+# Or if the we have a NODE_INSTANCE_PREFIX and the provider is gce
+# then run this code block.
+# 
+# This logic allows for the NODE_INSTANCE_GROUP to be set before this script runs, 
+# allowing for installers other than kube-up.sh to run GCE e2e tests. This also
+# allows for additional functionality to be added to kubetest to back out these
+# bash calls.
+#   
+# Note below is copied from the TODO when using the gke cloud provider.
+#
+# TODO(kubernetes/test-infra#3330): Allow NODE_INSTANCE_GROUP to be
+# set before we get here, which eliminates any cluster/gce use if
+# KUBERNETES_CONFORMANCE_PROVIDER is set to "gce".
+if [[ -z "${NODE_INSTANCE_GROUP:-}" && "${KUBERNETES_PROVIDER}" == "gce" ]] || [[ -n "${NODE_INSTANCE_PREFIX:-}" && "${KUBERNETES_PROVIDER}" == "gce" ]]; then
```

Comment:
The goal here is to get rid of this kind of logic, and have it set directly in kubetest or by the kubetest caller.

Is there a way we can have a flag/var for "do not set `NODE_INSTANCE_GROUP or run set_num_migs`, then first configure the kops jobs to use this and slowly migrate all the jobs to set NODE_INSTANCE_GROUP manually?

Reply:
@justinsb did this https://github.com/kubernetes/test-infra/pull/7408 as well.  The problem I have had is with testing ... getting ginkgo to run has been a bundle of fun :(

Reply:
> Is there a way we can have a flag/var for "do not set NODE_INSTANCE_GROUP or run set_num_migs, then first configure the kops jobs to use this and slowly migrate all the jobs to set NODE_INSTANCE_GROUP manually?

That is what I am basically doing ... making this code block optional and having kubetest sets the logic.  As I mentioned testing this is interesting.  Do we have instructions on how to run a custom bazel build test?  I opened an issue in test-infra https://github.com/kubernetes/test-infra/issues/7404 - because I am having issues testing this as well as I would like.

Reply:
I was thinking of something more explicit like:
```
if [[ -n ${MANUAL_GROUP_CONFIG:-} ]]; then
  echo "You are an awesome person"
else
  echo "DEPRECATED, PLEASE TODO BLAH BLAH"
  if [[ "${KUBERNETES_PROVIDER}" == "gce" ]]; then
    set_num_migs
    for ((i=1; i<${NUM_MIGS}; i++); do
  ...
  fi
fi
```

Rather than trying to infer this based on whether NODE_INSTANCE_FOO is already set...

And then setting `MANUAL_GROUP_CONFIG` for kops and sending test-infra a bug to fix the rest of the jobs.

Does this pattern have any merit?

Reply:
I thought about that approach as well.  Because we have two if blocks that interact, and since we have GKE code that has been modified in the same manner, I choose to model the pattern before.

I have no opinion either way what is better, both work.  What is more maintainable is probably the best question.

Reply:
Personally I would prefer to an early abort out of the sh logic rather than to add more sh complexity